js/kucoin.js

```diff
@@ -1651,6 +1655,37 @@ module.exports = class kucoin extends Exchange {
         }, market);
     }
 
+    async fetchTradingFee (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbols': market['id'],
+        };
+        const response = await this.privateGetTradeFees (this.extend (request, params));
+        // {
+        //     code: '200000',
+        //     data: [
+        //       {
+        //         symbol: 'BTC-USDT',
+        //         takerFeeRate: '0.001',
+        //         makerFeeRate: '0.001'
+        //       }
+        //     ]
+        // }
+        const data = this.safeValue (response, 'data', [ {} ]);
+        const fee = data[0];
+        const feeSymbol = this.safeString ('symbol', fee);
```

Comment:
Line 1677 is wrong, the first argument to safeString is the dictionary or array to search in, and the second argument is the field name to search for.
Also, let's rename the `feeSymbol` to `marketId`, we use that name everywhere.

```JavaScript
        const feeSymbol = this.safeString ('symbol', fee);
```
↓
```JavaScript
        const marketId = this.safeString (first, 'symbol');
        const symbol = this.safeSymbol (marketId, market);
        return {
            'info': response,
            'symbol': symbol,
            ...
        }
```

Reply:
I made the change with the exception I kept safeSymbol as a name because symbol is already declared as the function argument.